carbon-design-system/carbon
# feat(HeaderMenu):  include isCurrentPage props to HeaderMenu component

Closes #9784 

Currently, the links of a HeaderNavigation component (HeaderMenuItem) already have this behavior of Convey menu items and their states setting border-bottom style. Example: 

<img width="510" alt="tday" src="https://user-images.githubusercontent.com/8935295/135544709-15f904b3-6246-4e58-a18c-4c4455d834f1.png">

The point is that when creating a dropdown menu, I mean a HeaderMenu with HeaderMenuItems as children, like in "link 4" in the image above, there isn't a prop expected to set it as an active menu. 

There is already a way to set the items inside the HeaderMenu as Active like below: 
<img width="295" alt="menu-open" src="https://user-images.githubusercontent.com/8935295/135545014-793112cf-eb4f-4cdb-b08e-a1187deb96fe.png">

As described in one item of [W3 WAI (Web Accessibility Tutorials)](https://www.w3.org/WAI/tutorials/menus/styling/) it's important to:
"Convey menu items and their states by using color and other styling options. For example, alter the shape of a menu item, or add an icon, in addition to changing its color when it is selected."

When the HeaderMenu is closed and a page inside is active the user has no signal of wich page is select looking at the menu. 

Before 
<img width="473" alt="4select-n" src="https://user-images.githubusercontent.com/8935295/135545366-ebfc6cd9-98c2-40cc-88e5-00de8564f35b.png">

After
<img width="510" alt="4selected" src="https://user-images.githubusercontent.com/8935295/135545416-ef7e69ef-0801-4ef5-89e3-a793eaf5d409.png"


###

# Changelog

**New**

- added inside the `carbon/packages/components/src/components/ui-shell/_header.scss` and the `carbon/packages/styles/scss/components/ui-shell/header/_header.scss` a new class (${prefix}--header__submenu--current) to set the active border style (same style as HeaderMenuItem).


**Changed**
- ClassName const inside HeaderMenu to link the new class created, when the prop isCurrentPage is equal true


#### Testing / Reviewing

- Run react storybook
- Go in the file `react/src/components/UIShell/UIShell-story.js`
- Put the prop isCurrentPage in the HeaderMenu component tag. Example: 
```javascript 
<HeaderMenu aria-label="Link 4" menuLinkName="Link 4" isCurrentPage>
```
- Ensure that the border was added to the bottom of the HeaderMenu on the storybook  "Header Base w/ Navigation" example
